python/pyspark/sql/worker/plan_data_source_read.py

```diff
@@ -54,17 +54,31 @@
 
 
 def records_to_arrow_batches(
```

Comment:
This method is also used by streaming reader. cc @chaoqin-li1123 do you think we should support arrow batch output for streaming reader as well?

Reply:
I don't have a strong opinion here. For API consistency I prefer to make batch and streaming data source has the same return type.